test/client/fixtures/sandbox/cookie-test.js

```diff
@@ -210,6 +211,152 @@ if (window.fetch) {
     });
 }
 
+module('synchronization between frames');
+
+test('same-domain frames', function () {
+    var sameDomainSrc  = getSameDomainPageUrl('../../data/cookie-sandbox/same-domain-iframe.html');
+    var iframe         = null;
+    var embeddedIframe = null;
+
+    return createTestIframe({ src: sameDomainSrc })
+        .then(function (createdIframe) {
+            iframe = createdIframe;
+
+            return createTestIframe({ src: sameDomainSrc + '&iframe=embedded' }, iframe.contentDocument.body);
+        })
+        .then(function (createdIframe) {
+            embeddedIframe = createdIframe;
+
+            function checkCookies (expectedCookies) {
+                strictEqual(nativeMethods.documentCookieGetter.call(document), '');
+                strictEqual(settings.get().cookie, expectedCookies);
+                strictEqual(iframe.contentWindow['%hammerhead%'].get('./settings').get().cookie, expectedCookies);
+                strictEqual(embeddedIframe.contentWindow['%hammerhead%'].get('./settings').get().cookie, expectedCookies);
+            }
+
+            checkCookies('');
+
+            nativeMethods.documentCookieSetter.call(document, 's|sessionId|test|example.com|%2F||1fckm5lnl=123;path=/');
+            nativeMethods.documentCookieSetter.call(document, 's|sessionId|cafe|example.com|%2F||1fckm5lnl=321;path=/');
+
+            document.cookie; // eslint-disable-line no-unused-expressions
+
+            checkCookies(browserUtils.isSafari ? 'cafe=321; test=123' : 'test=123; cafe=321');
```

Comment:
It looks like that Safari sorts cookie by name.
I propose to change order of cookie set with native setter to reduce expression for expected value.